pkg/cmd/edge_applications/list/list.go

```diff
@@ -0,0 +1,108 @@
+package list
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/fatih/color"
+
+	"github.com/MakeNowJust/heredoc"
+	table "github.com/MaxwelMazur/tablecli"
+	msg "github.com/aziontech/azion-cli/messages/edge_applications"
+	api "github.com/aziontech/azion-cli/pkg/api/edge_applications"
+	"github.com/aziontech/azion-cli/pkg/cmdutil"
+	"github.com/aziontech/azion-cli/pkg/contracts"
+	"github.com/spf13/cobra"
+)
+
+func NewCmd(f *cmdutil.Factory) *cobra.Command {
+	opts := &contracts.ListOptions{}
+	cmd := &cobra.Command{
+		Use:           msg.EdgeApplicationsListUsage,
+		Short:         msg.EdgeApplicationsListShortDescription,
+		Long:          msg.EdgeApplicationsListLongDescription,
+		SilenceUsage:  true,
+		SilenceErrors: true, Example: heredoc.Doc(`
+		$ azioncli edge_applications list --details
+		$ azioncli edge_applications ist --order_by "id"
+		$ azioncli edge_applications list --page 1  
+		$ azioncli edge_applications list --page_size 5
+		$ azioncli edge_applications list --sort "asc" 
+		`),
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var numberPage int64 = opts.Page
+			if !insertTheFlagPage(cmd) {
+				for {
+					pages, err := PrintTable(cmd, f, opts, &numberPage)
+					if numberPage > pages && err == nil {
+						return nil
+					}
+					if err != nil {
+						return fmt.Errorf(msg.ErrorGetApplication.Error(), err)
+					}
+				}
+			}
+
+			if _, err := PrintTable(cmd, f, opts, &numberPage); err != nil {
+				return fmt.Errorf(msg.ErrorGetApplication.Error(), err)
+			}
+			return nil
+		},
+	}
+
+	cmdutil.AddAzionApiFlags(cmd, opts)
+	cmd.Flags().BoolP("help", "h", false, msg.EdgeApplicationsListHelpFlag)
+	return cmd
+}
+
+func insertTheFlagDetails(cmd *cobra.Command) bool {
```

Comment:
if you really want to create a function for checking if the flag is present, you can do it generically, like you did on init command. 

`func hasThisFlag(c *cobra.Command, flag string) bool {
	return c.Flags().Changed(flag)
}`

Reply:
A ideia é só deixar mais legivel, acredito que fique bom e legivel